doc/tutorial/install.rst

```diff
@@ -40,3 +40,24 @@ nix
 `Nix <https://nixos.org/nix/>`_ users should feel free to take a look at
 the `nix/shell.nix` file in the repository and use it to provision a suitable
 environment to build and run the examples.
+
+Note for Ubuntu users
+--------
+
+Ubuntu's packages for `ghc`, `cabal`, and `stack` are years out of date.
+If the instructions above fail for you,
+try replacing the Ubuntu packages with up-to-date versions.
+First remove the installed versions:
+
+.. code-block:: bash
+
+   # remove the obsolete versions
+   $ sudo apt remove ghc haskell-stack cabal-install
+
+Then install fresh versions of the Haskell toolchain
+using the `ghcup <https://www.haskell.org/ghcup/install/>`_ installer.
+As of February 2022, an easy way to do this is by running a bootstrap script:
+
+.. code-block:: bash
+
+   $ curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | sh
```

Comment:
I'd either leave out all commands and point to some official ghcup instructions, or additionally include one that shows how to install `ghc` and `¢abal`/`stack` when `ghcup` is installed. This seems to sit between the two approaches without that last final step =)

Reply:
I considered both of those options and chose this specifically.  Regarding #1, I really dislike instructions that say “now go off and perform an unknown number of operations as directed by a completely different web site”; I like to get some idea of what is coming next and whether there is an easy option.  Often the target web site has been reorganized and it's not clear where to find further details.

Regarding #2, the indicated installer is conversational.  It prompts the user for what to do next, so there's no reason to try to embed its output here.  Any such embedded output would be soon out of date.

Perhaps it would suffice to add a single sentence that explains “from there the installation process is interactive”.

Reply:
Yes that final sentence would be very helpful I think, I had forgotten the interactive part with `ghcup\ even though I've already used it a few times, for example =)

Thanks!

Reply:
Updated per your suggestion.